src/main/java/com/epam/Main.java

```diff
@@ -1,24 +1,171 @@
+        	batchSize = DEFAULT_BATCH_SIZE;
+        }
+        
+        int producerSize = 0;
+        try{
+        	producerSize = Integer.parseInt(args[3]);
+        }catch(Exception e){
+        	producerSize = DEFAULT_PRODUCER_SIZE; 
+        }
+        
+        int consumerSize = 0;
+        try{
+        	consumerSize = Integer.parseInt(args[4]);
+        }catch(Exception e){
+        	consumerSize = DEFAULT_CONSUMER_SIZE; 
+        }
+        handleMultipleFilesWithCallable(inputFileQueue,threadPoolSize,batchSize,producerSize,consumerSize);
+    }
+    
+    public static void handleMultipleFilesWithCallable(LinkedList<String> inputFileQueue,int threadPoolSize, int batchSize,int producerSize, int consumerSize) throws ExecutionException, InterruptedException, IOException {
+        
+        ExecutorService executor = Executors.newFixedThreadPool(threadPoolSize);
 
-    private static final String ACCIDENTS_CSV = "src/main/resources/DfTRoadSafety_Accidents_2009.csv";
-
-
-    public static void main(String[] args) throws IOException {
-        AccidentsDataLoader accidentsDataLoader = new AccidentsDataLoader();
-        List<RoadAccident> accidents = accidentsDataLoader.loadRoadAccidents(ACCIDENTS_CSV);
+        BlockingQueue<List<RoadAccident>> dataQueue = new ArrayBlockingQueue<List<RoadAccident>>(DEFAULT_STORAGE_QUEUE_SIZE);        
+        FileWriter fileWriter1 = new FileWriter(OUTPUT_FILE_PATH_DAYTIME);
+        CSVPrinter csvWriterDayTime = new CSVPrinter(fileWriter1,csvFormat);
+        printCSVHeader(csvWriterDayTime);
+        
+        FileWriter fileWriter2 = new FileWriter(OUTPUT_FILE_PATH_NIGHTTIME);
+        CSVPrinter csvWriterNightTime = new CSVPrinter(fileWriter2,csvFormat);
+        printCSVHeader(csvWriterNightTime);
+        //start consumer threads
+        System.out.println("Starting Consumers");
+        for(int i=0;i<consumerSize;i++){
+        	executor.execute(new DataCSVFileWriter(dataQueue, csvWriterDayTime,csvWriterNightTime));
+        }    
+            
+       
 
+        //Create 2 reader task and start
+        System.out.println("Starting readers");
+        List<FutureTask<Integer>> futureTaskList = new ArrayList<FutureTask<Integer>>(); 
+        
+        for (int i = 0; i < producerSize; i++) {
+			if (inputFileQueue.isEmpty()) {
+				Thread.sleep(1000*5);
+				break;
+			}
+			FutureTask<Integer> readerTask = new FutureTask<Integer>(
+					new AccidentBatchLoader(batchSize, dataQueue,
+							inputFileQueue.remove()));
+			futureTaskList.add(readerTask);
+			executor.execute(readerTask);
 
+		}      
+               
+        while(!inputFileQueue.isEmpty()){
+        	List<FutureTask<Integer>> newFutureTaskList = new ArrayList<FutureTask<Integer>>();
+        	for(FutureTask<Integer> task:futureTaskList){
+        		if(task.isDone()){
+        			if (inputFileQueue.isEmpty()) {
+        				break;
+        			}
+        			FutureTask<Integer> readerTask = new FutureTask<Integer>(
```

Comment:
Ohh! I got it - quite clever approach to limit a load. 

You know why I am confused? Clean code! If you separate logic in methods with meaningful names - like submitJobsWithLimit or somethign - then it would take less time for me to understand it :smirk: